components/sup/src/event/ratsio.rs

```diff
@@ -55,15 +56,22 @@ pub(super) fn init_stream(conn_info: EventConnectionInfo) -> Result<EventStream>
                                              .expect("Couldn't synchronize event thread!");
 
                                       event_rx.for_each(move |event: Vec<u8>| {
-                                          let stan_msg =
-                                              StanMessage::new(HABITAT_SUBJECT.into(),
-                                                               event);
-                                          let publish_event = client
-                                              .send(stan_msg)
-                                              .map_err(|e| {
-                                                  error!("Error publishing event: {}", e)
-                                              });
-                                          executor::spawn(publish_event);
+                                          if client.nats_client.get_state() == NatsClientState::Connected {
+                                              let stan_msg =
+                                                  StanMessage::new(HABITAT_SUBJECT.into(),
+                                                                   event);
+                                              let publish_event = client
+                                                  .send(stan_msg)
+                                                  .map_err(|e| {
+                                                      error!("Error publishing event: {}", e)
+                                                  });
+                                              executor::spawn(publish_event);
+                                          } else {
+                                              trace!(
+                                                  "Unable to send event because client is in state {:?}",
+                                                  client.nats_client.get_state()
+                                              );
+                                          }
```

Comment:
Minor, but it would be nice to recast this `if`/`else` as a `match`, so you could reuse the state:
```rust
match client.nats_client.get_state() {
    NatsClientState::Connected => {
        // do normal stuff
    },
    state => {
        trace!("Unable to send event because client is in state {:?}", state);
    }
}
```

Reply:
A bit of a comment here about how we're intentionally dropping unsendable events on the floor while we're not connected to a NATS server (for now, at least), would be useful, so that any future refactorings could preserve that intent.